scss/utilities/_clearfix.scss

```diff
@@ -1,3 +1,11 @@
 .clearfix {
   @include clearfix();
 }
+
+@each $breakpoint in map-keys($grid-breakpoints) {
+  @include media-breakpoint-only($breakpoint) {
```

Comment:
If there are "un-clearfix" classes (by analogy to `.pull-*-none`), then it's possible to express the same thing as the "only" classes by combining "un-clearfix" classes with "up" classes.
I also question how representative your example is. IME, folks don't use all the grid tiers all the time. Most of the time they only use 2-3 of them, which would make having to add extra clearfix classes for the remaining unused grid tiers somewhat of a pain.
Thirdly, nothing in the framework itself currently uses `media-breakpoint-only`, so it's arguably more consistent to avoid it.

But I agree the choice either way is debatable. Both seem workable, it's just a matter of which "feels" better.
Think we'll need @mdo to weigh in on this.